server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/CassandraBlobsDAO.java

```diff
@@ -164,16 +163,13 @@ private Integer getChunkNum(Pair<Integer, ByteBuffer> pair) {
     }
 
     @Override
-    public CompletableFuture<byte[]> readBytes(BlobId blobId) {
+    public Mono<byte[]> readBytes(BlobId blobId) {
         try {
             return readBlobParts(blobId)
                 .collectList()
-                .map(parts -> Bytes.concat(parts.toArray(new byte[0][])))
-                .toFuture();
+                .map(parts -> Bytes.concat(parts.toArray(new byte[0][])));
         } catch (ObjectStoreException e) {
-            CompletableFuture<byte[]> error = new CompletableFuture<>();
-            error.completeExceptionally(e);
-            return error;
+            return Mono.error(e);
```

Comment:
Same as above, can maybe use the Mono error methods to get rid of the try catch blocks?

Reply:
Moreover, nothing should be executed at this point, so nothing should be thrown.